### ISSUE TITLE
Editor crashes after pasting multi-block content over existing content in a blockquote

### ISSUE THREAD

**Issue by Mgsy** [ckeditor/ckeditor5#1921]
## Is this a bug report or feature request? (choose one)

🐞 Bug report

## 💻 Version of CKEditor

Latest master

## 📋 Steps to reproduce

1. Set the following data: `editor.setData( '<blockquote><p>test</p><p>test</p></blockquote>' );`
1. Copy those paragraphs.
1. Press <kbd>Ctrl</kbd> + <kbd>A</kbd>.
1. Paste.

## ✅ Expected result
Content is pasted properly.

## ❎ Actual result
The editor crashes.

## 📃 Other details that might be useful
**Error**
```
schema.js:1156 Uncaught TypeError: Cannot read property 'getAncestors' of null
    at new me (schema.js:1156)
    at fe.on.priority (schema.js:58)
    at fe.fire (emittermixin.js:207)
    at fe.<computed> [as checkChild] (observablemixin.js:259)
    at fn._getAllowedIn (insertcontent.js:603)
    at fn._checkAndSplitToAllowedPosition (insertcontent.js:544)
    at fn._handleNode (insertcontent.js:259)
    at fn.handleNodes (insertcontent.js:185)
    at Object.callback (insertcontent.js:69)
    at xn._runPendingChanges (model.js:738)
```

**GIF**
![bug_cke5](https://user-images.githubusercontent.com/20988892/62042721-0af33a80-b1ff-11e9-947f-366065142e52.gif)

It's a regression. I've bisected it and it looks like these changes introduced a bug - https://github.com/ckeditor/ckeditor5-engine/commit/f4e46447100335ec08d8d54b77215718b74724c5

**Comment by pomek** [comment]
I wrote a test that reproduces the issue:

```js
it( 'does not blow up after pasting items over block-quote', () => {
	setModelData( model,
		'[<blockQuote>' +
			'<paragraph>Foo</paragraph>' +
			'<paragraph>Bar</paragraph>' +
		'</blockQuote>]'
	);

	console.log( getModelData( model ) );

	pasteHtml( editor, '<p>FooBar</p>' );

	console.log( getModelData( model ) );
} );

function pasteHtml( editor, html ) {
	editor.editing.view.document.fire( 'paste', {
		dataTransfer: createDataTransfer( { 'text/html': html } ),
		preventDefault() {
		}
	} );
}

function createDataTransfer( data ) {
	return {
		getData( type ) {
			return data[ type ];
		}
	};
}
```

And it fails (as expected):

```
CKEditorError: model-position-path-incorrect: The position's path is incorrect. Read more: https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-model-position-path-incorrect
       {"position":{"root":"main","path":[0,0,0],"stickiness":"toNext"}}
```

A place in the code where the error was thrown: https://github.com/ckeditor/ckeditor5-engine/blob/04fa92058a83d70c7384713cab9f462d2f1ae90f/src/model/utils/insertcontent.js#L139

And I guess, I understand why it happens. The position after removing all nodes from the editor is invalid. After switching the position to be an instance of  `LivePosition`, the error does not occur. But it isn't a solution. That gave me an idea of what could be wrong.

Then, I decided to write one more test for the `insertcontent` util that does more or less the same as the previous one:

```js
it.only( 'deletes selection with elements before inserting the content', () => {
	model.schema.register( 'p', { inheritAllFrom: '$block' } );
	model.schema.register( 'div', {
		allowWhere: '$block',
		allowContentOf: '$root'
	} );

	setData( model, '[<div><p>Foo</p><p>Bar</p></div>]' );
	const affectedRange = insertHelper( '<p>FooBar</p>' );

	expect( getData( model ) ).to.equal( '<p>FooBar</p>[]' );
	expect( stringify( root, affectedRange ) ).to.equal( '[<p>FooBar</p>]' );
} );
```

Unfortunately, it didn't fail as I expected. For such case, a position used to create an instance of `Insertion` class seems to be valid.

What am I doing wrong here?

**Comment by pomek** [comment]
Thanks to @Reinmar, we figured out what was wrong with the second test. Instead of using a short name of paragraph (`p`), I should use a full name of the element because the engine assumes that the element exists.

The corrected tests should look like:

```js
it( 'deletes selection with elements before inserting the content', () => {
	model.schema.register( 'div', {
		allowWhere: '$block',
		allowContentOf: '$root'
	} );

	setData( model, '<div><paragraph>[Foo</paragraph><paragraph>Bar]</paragraph></div>' );
	const affectedRange = insertHelper( '<paragraph>FooBar</paragraph>' );

	expect( getData( model ) ).to.equal( '<paragraph>FooBar[]</paragraph>' );
	expect( stringify( root, affectedRange ) ).to.equal( '[<paragraph>FooBar</paragraph>]' );
} );
```

And this one, as expected, fails. Going into deep, it leads to this issue – https://github.com/ckeditor/ckeditor5-engine/issues/1722 and I'll continue posting my thoughts here.

**Comment by pomek** [comment]
Using the [Title](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/features/title.html) plugin, the issue is even "easier" to reproduce.

Type whatever in the title section, then select all (CTRL/CMD+A) then paste anything.

**Comment by Reinmar** [comment]
DUP of #2010. Fixed with [ckeditor/ckeditor5-engine#1830](https://github.com/ckeditor/ckeditor5-engine/pull/1830).
